tests/ut/backends/test_valkey.py

```diff
@@ -240,3 +240,41 @@ def test_build_key_double_dot(
 
     def test_build_key_no_namespace(self, valkey_cache):
         assert valkey_cache.build_key(Keys.KEY, namespace=None) == Keys.KEY
+
+    def test_custom_serializer(self, valkey_config):
+        assert isinstance(
+            ValkeyCache(config=valkey_config, serializer=PickleSerializer()).serializer,
+            PickleSerializer,
+        )
+
+    def test_default_key_builder(self, valkey_config):
+        real_code = ValkeyCache(config=valkey_config)._build_key.__code__
+        expected_code = (lambda k, ns: f"{ns}:{k}" if ns else k).__code__
+        assert (
+            real_code.co_code == expected_code.co_code
+            and real_code.co_consts == expected_code.co_consts
+            and real_code.co_names == expected_code.co_names
+        )
```

Comment:
I feel like a cleaner way to test this would be to provide a different key builder and test a call produces the expected result.

e.g. `lambda k, ns: f"{ns}__k" if ns else k`
Then verify that using the cache with a namespace saves with the `{ns}__k` key.